subsys/llext/llext_load.c

```diff
@@ -630,6 +692,13 @@ int do_llext_load(struct llext_loader *ldr, struct llext *ext,
 		goto out;
 	}
 
+	LOG_DBG("Calling init functions...");
+	ret = llext_call_inits(ldr, ext);
```

Comment:
This is changing the loading behavior from simple parse and copy elf data to actually calling functions in the elf file.

I'm concerned this leaves a pretty large gap for untrusted code to inject malicious behavior and a user of llext may never know it.

Maybe its worth reviewing this behavior @ceolin

Reply:
Yep, the elf is assumed to be untrusted by now (since there is no authenticity / integrity check). Execute these functions in the kernel context is definitely a problem. Can you defer this initialization for after the elf is loaded, and the user be responsible to call it assuming it is untrusted ?

Reply:
Sure, this is a valid point. I will make the llext_call_inits function part of the API so the user will need to explicitly  call it if it's needed.

In the future however I would like to find a way to make this process automatic, like it is done for Linux modules. Definitely needs at least some kind of validation and an "llext header" describing Zephyr related features (such as should the initi function be user or kernel space), though.

Reply:
FWIW SOF currently does sign its LLEXT objects and verifies signature correctness. If Zephyr decides to implement an API for this, SOF will probably also use it

Reply:
@lyakh I think Zephyr should provide infrastructure to that, there are too many caveats and pretty much everyone using LLEXT in products will need it.

Reply:
@pillo79 The Linux model is not exactly the same as here. Linux won't be blindly executing an userspace application in kernel context. If the module is trusted we may do how is proposed here, but we have to cover the untrusted case.
Maybe for untrusted modules we can sandbox the module before initializing it ?